Failed at the cdk-sample@0.1.0 build script

I have an issue when try to build:
```
$ aws-rails-provisioner build                                                                                                                                                                                          
replacing cdk-sample/lib/cdk-sample-init-stack.ts
replacing cdk-sample/lib/aws-rails-provisioner-fargate-stack.ts
replacing cdk-sample/bin/cdk-sample.ts
npm WARN cdk-sample@0.1.0 No repository field.
npm WARN cdk-sample@0.1.0 No license field.

npm WARN cdk-sample@0.1.0 No repository field.
npm WARN cdk-sample@0.1.0 No license field.

npm WARN cdk-sample@0.1.0 No repository field.
npm WARN cdk-sample@0.1.0 No license field.

npm WARN cdk-sample@0.1.0 No repository field.
npm WARN cdk-sample@0.1.0 No license field.

npm WARN cdk-sample@0.1.0 No repository field.
npm WARN cdk-sample@0.1.0 No license field.

npm WARN cdk-sample@0.1.0 No repository field.
npm WARN cdk-sample@0.1.0 No license field.

running npm run build ...

npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! cdk-sample@0.1.0 build: `tsc`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the cdk-sample@0.1.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/samir/.npm/_logs/2020-02-05T11_24_57_799Z-debug.log
```
the log file:
```
0 info it worked if it ends with ok
1 verbose cli [
1 verbose cli   '/usr/local/Cellar/node/12.12.0/bin/node',
1 verbose cli   '/usr/local/bin/npm',
1 verbose cli   'run',
1 verbose cli   'build'
1 verbose cli ]
2 info using npm@6.11.3
3 info using node@v12.12.0
4 verbose run-script [ 'prebuild', 'build', 'postbuild' ]
5 info lifecycle cdk-sample@0.1.0~prebuild: cdk-sample@0.1.0
6 info lifecycle cdk-sample@0.1.0~build: cdk-sample@0.1.0
7 verbose lifecycle cdk-sample@0.1.0~build: unsafe-perm in lifecycle true
8 verbose lifecycle cdk-sample@0.1.0~build: PATH: /usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/Users/samir/Documents/projects/cdk-sample/node_modules/.bin:/Users/samir/.rvm/gems/ruby-2.7.0/bin:/Users/samir/.rvm/gems/ruby-2.7.0@global/bin:/Users/samir/.rvm/rubies/ruby-2.7.0/bin:/Users/samir/.rvm/bin:/usr/local/opt/dart@1/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/share/dotnet:~/.dotnet/tools:/Library/Frameworks/Mono.framework/Versions/Current/Commands:/Users/samir/Library/Android/sdk/platform-tools
9 verbose lifecycle cdk-sample@0.1.0~build: CWD: /Users/samir/Documents/projects/cdk-sample
10 silly lifecycle cdk-sample@0.1.0~build: Args: [ '-c', 'tsc' ]
11 silly lifecycle cdk-sample@0.1.0~build: Returned: code: 2  signal: null
12 info lifecycle cdk-sample@0.1.0~build: Failed to exec build script
13 verbose stack Error: cdk-sample@0.1.0 build: `tsc`
13 verbose stack Exit status 2
13 verbose stack     at EventEmitter.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:332:16)
13 verbose stack     at EventEmitter.emit (events.js:210:5)
13 verbose stack     at ChildProcess.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:210:5)
13 verbose stack     at maybeClose (internal/child_process.js:1021:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:283:5)
14 verbose pkgid cdk-sample@0.1.0
15 verbose cwd /Users/samir/Documents/projects/cdk-sample
16 verbose Darwin 19.2.0
17 verbose argv "/usr/local/Cellar/node/12.12.0/bin/node" "/usr/local/bin/npm" "run" "build"
18 verbose node v12.12.0
19 verbose npm  v6.11.3
20 error code ELIFECYCLE
21 error errno 2
22 error cdk-sample@0.1.0 build: `tsc`
22 error Exit status 2
23 error Failed at the cdk-sample@0.1.0 build script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 2, true ]
```
settings file:

```
  1 version: "0"
  2 services: 
  3   aws-rails-provisioner:
  4     source_path: ./docker-test
  5     fargate: 
  6       desired_count: 3
  7       envs: 
  8         AWS_FARGATE_ENV_VAR: "Passed By aws-rails-provisioner"
  9         PORT: 80
 10         RAILS_LOG_TO_STDOUT: true
 11       public: true
 12 vpc:  
 13   cidr: 10.0.0.0/21
 14   enable_dns: true
 15   max_azs: 2

```

--- a/README.md
+++ b/README.md
@@ -4,10 +4,7 @@
 
 A tool for defining and deploying containerized Ruby on Rails applications on AWS.
 
-`aws-rails-provisioner` is a command line tool using your configurations defined in `aws-rails-provisioner.yml` file to generate 
-[AWS CDK](https://github.com/awslabs/aws-cdk) stacks on your behalf, provisioning required AWS resources
-for running your containerized Ruby on Rails applications on AWS (current supported platform: AWS Fargate) within
-few commands.
+`aws-rails-provisioner` is a command line tool using your configurations defined in `aws-rails-provisioner.yml` file to generate [AWS CDK](https://github.com/awslabs/aws-cdk) stacks on your behalf, provisioning required AWS resources for running your containerized Ruby on Rails applications on AWS (current supported platform: AWS Fargate) within a few commands.
 
 <!--BEGIN STABILITY BANNER-->
 ---
@@ -27,9 +24,9 @@ few commands.
 
 ### Prerequisites
 
-Before using the `aws-rails-provisioner` gem, you need to have 
+Before using the `aws-rails-provisioner` gem, you need to have:
 
-* a [Ruby on Rails](https://rubyonrails.org/) application with Dockerfile 
+* a [Ruby on Rails](https://rubyonrails.org/) application with a Dockerfile
 * install or update the [AWS CDK Toolkit] from npm (requires [Node.js ≥ 8.11.x](https://nodejs.org/en/download)):
 
   ```bash
@@ -47,8 +44,7 @@ gem install aws-rails-provisioner --pre
 
 ### Define aws-rails-provisioner.yml
 
-`aws-rails-provisioner.yml` is a configuration file defining how `aws-rails-provisioner` bootstraps required AWS resources
-for your Ruby on Rails applications.
+`aws-rails-provisioner.yml` is a configuration file defining how `aws-rails-provisioner` bootstraps required AWS resources for your Ruby on Rails applications.
 
 ```
 version: '0'
@@ -80,33 +76,30 @@ services:
   my_another_rails:
     ...
 ```
-More `aws-rails-provisioner` examples see `./examples` (see `tiny.yml` for a minimal `aws-rails-provisioner.yml` configuration example),
-full configuration options documentation see [here](https://docs.aws.amazon.com/awsrailsprovisioner/api/).
+
+See `./examples` for more `aws-rails-provisioner` examples (see `tiny.yml` for a minimal `aws-rails-provisioner.yml` configuration example). The full configuration options are documented [here](https://docs.aws.amazon.com/awsrailsprovisioner/api/).
 
 ### Build and Deploy
 
-Once `aws-rails-provisioner.yml` is defined, run build command as following will boostrap AWS CDK stacks in code
-defining all required AWS resources and connections.
+Once `aws-rails-provisioner.yml` is defined, run the build command. This will boostrap AWS CDK stacks and define all required AWS resources and connections.
 
 ```
 aws-rails-provisioner build
 ```
 
-By default, it defines a VPC with public and private subnets, Amazon RDS Database Cluster, an ECS cluster with
-AWS Fargate services containing application images, when run build with `--with-cicd` option as following
-an CICD stack will be defined automatically (including data migration step)
+By default, it defines a VPC with public and private subnets, an Amazon RDS Database Cluster, an ECS cluster with AWS Fargate services containing application images. When building with `--with-cicd` option a CICD stack will be defined automatically (including a Rails data migration step).
 
 ```
 aws-rails-provisioner build --with-cicd
 ```
 
-After build complete, run the deploy command to deploy all defined AWS resources
+After the build completes, run the deploy command to deploy all defined AWS resources:
 
 ```
 aws-rails-provisioner deploy
 ```
 
-Instead of deploy everything all at once, you can deploy stack by stack, application by application
+Instead of deploy everything all at once, you can deploy stack by stack, application by application:
 
 ```
 # only deploys the stack creates VPC and ECS cluster
@@ -119,21 +112,16 @@ aws-rails-provisioner deploy --fargate
 aws-rails-provisioner deploy --cicd
 
 # deploy only :rails_foo application
-aws-rails-provisioner deploy --fargate --service rails_foo
+aws-rails-provisioner deploy --fargate --service my_rails_foo
 ```
 
-After deployment completes, your applications are running on AWS Fargate fronted with AWS Application
-LoadBalancer.
+After the deployment completes, your applications are now running on AWS Fargate fronted with AWS Application LoadBalancer. You can view your website using the public IP address of the network interface's load balancer.
 
-> Note: for applications with databases, rails migration is needed, CICD stack contains a migration phase
-by default, running DB migration commands insides private subnets, talking to DB Cluster.
+> Note: for applications with databases, rails db migration is needed; the CICD stack contains a migration phase by default, running DB migration commands insides private subnets, connected to the DB Cluster.
 
 ### CICD
 
-When have `--with-cicd` enabled at build, a CICD stack is available. Once deployment completes, an AWS
-CodePipeline is available with source, build, migration and deploy phases. You need to commit
-your local Rails application to the source repository in the pipeline with `buildspec` to activate the pipeline.
-Sample `buildspec`s are availble under `./buildspecs` handling application image build and rails migration.
+When `--with-cicd` is enabled at build, a CICD stack is created. Once deployment completes, an AWS CodePipeline is available with source, build, migration, and deploy phases. You need to commit your local Rails application to the CodeCommit source repository in the pipeline with `buildspec` files to activate the pipeline. Sample `buildspec`s are availble under `./buildspecs` handling application image builds and rails migrations.
 
 Full `aws-rails-provisioner` command line options see:
 
@@ -147,14 +135,11 @@ We welcome community contributions and pull requests. See [CONTRIBUTING](./CONTR
 
 ## Getting Help
 
-Please use these community resources for getting help. We use the GitHub issues
-for tracking bugs and feature requests.
+Please use these community resources for getting help. We use the GitHub issues for tracking bugs and feature requests.
 
-* Ask a question on [Stack Overflow](https://stackoverflow.com/questions/tagged/aws-rails-provisioner)
-    and tag it with `aws-rails-provisioner`
+* Ask a question on [Stack Overflow](https://stackoverflow.com/questions/tagged/aws-rails-provisioner) and tag it with `aws-rails-provisioner`
 * If it turns out that you may have found a bug, or want to submit a feature request, please open an [issue](https://github.com/awslabs/aws-cdk/issues/new)
 
 ## License
 
-The `aws-rails-provisioner` is distributed under [Apache License, Version 2.0](https://www.apache.org/licenses/LICENSE-2.0).
-See [LICENSE](./LICENSE.txt) for more information.
+The `aws-rails-provisioner` is distributed under [Apache License, Version 2.0](https://www.apache.org/licenses/LICENSE-2.0). See [LICENSE](./LICENSE.txt) for more information.

--- a/README.md
+++ b/README.md
@@ -81,7 +81,7 @@ See `./examples` for more `aws-rails-provisioner` examples (see `tiny.yml` for a
 
 ### Build and Deploy
 
-Once `aws-rails-provisioner.yml` is defined, run the build command. This will boostrap AWS CDK stacks and define all required AWS resources and connections.
+Once `aws-rails-provisioner.yml` is defined, run the build command. This will boostrap AWS CDK stacks and define all required AWS resources and connections in CDK code.
 
 ```
 aws-rails-provisioner build
@@ -93,13 +93,13 @@ By default, it defines a VPC with public and private subnets, an Amazon RDS Data
 aws-rails-provisioner build --with-cicd
 ```
 
-After the build completes, run the deploy command to deploy all defined AWS resources:
+After the build completes, run the deploy command to run CDK code that deploys all defined AWS resources:
 
 ```
 aws-rails-provisioner deploy
 ```
 
-Instead of deploy everything all at once, you can deploy stack by stack, application by application:
+Instead of deploying everything all at once, you can deploy stack by stack, application by application:
 
 ```
 # only deploys the stack creates VPC and ECS cluster
@@ -121,7 +121,7 @@ After the deployment completes, your applications are now running on AWS Fargate
 
 ### CICD
 
-When `--with-cicd` is enabled at build, a CICD stack is created. Once deployment completes, an AWS CodePipeline is available with source, build, migration, and deploy phases. You need to commit your local Rails application to the CodeCommit source repository in the pipeline with `buildspec` files to activate the pipeline. Sample `buildspec`s are availble under `./buildspecs` handling application image builds and rails migrations.
+When `--with-cicd` is enabled at build, a CICD stack is created. Once deployment completes, an AWS CodePipeline is available with source, build, migration, and deploy phases for your Rails application. You need to commit your Rails application to the CodeCommit source repository in the pipeline with `buildspec` files to activate the pipeline. Sample `buildspec`s are availble under `./buildspecs` handling application image builds and rails migrations.
 
 Full `aws-rails-provisioner` command line options see:
 

--- a/lib/aws-rails-provisioner/cdk_builder.rb
+++ b/lib/aws-rails-provisioner/cdk_builder.rb
@@ -56,6 +56,7 @@ module Aws::RailsProvisioner
       end
       Dir.chdir(@cdk_dir) do
         pkgs.each do |pkg|
+          puts "installing pkg: #{pkg} ..."
           `npm install #{pkg}`
         end
       end
@@ -64,7 +65,7 @@ module Aws::RailsProvisioner
     def _npm_build
       Dir.chdir(@cdk_dir) do
         puts "running npm run build ..."
-        `npm run build`
+        `npm run build --verbose`
       end
     end
 

--- a/lib/aws-rails-provisioner/cdk_deployer.rb
+++ b/lib/aws-rails-provisioner/cdk_deployer.rb
@@ -20,6 +20,7 @@ module Aws::RailsProvisioner
 
     def run
       Dir.chdir(@cdk_dir) do
+        `cdk bootstrap`
         deploy_init_stack
         unless @init_stack_only
           if @svc_name

--- a/lib/aws-rails-provisioner/db_cluster.rb
+++ b/lib/aws-rails-provisioner/db_cluster.rb
@@ -1,6 +1,6 @@
 require 'aws-sdk-rds'
 
-module Aws::RailsProvisioner 
+module Aws::RailsProvisioner
   class DBCluster
 
     # Configuration value under :db_cluster
@@ -59,9 +59,9 @@ module Aws::RailsProvisioner
     #
     #   @example: at `aws-rails-provisioner.yml`
     #     parameter_group:
-    #       family: 'aurora-postgresql9.6'
+    #       family: 'aurora-postgresql11'
     #       description: 'created by AWS RailsProvisioner'
-    #       parameters: 
+    #       parameters:
     #         key: value
     #   @see {Aws::RailsProvisioner::DBCluster::ParameterGroup}
     #
@@ -178,7 +178,7 @@ module Aws::RailsProvisioner
     end
 
     class ParameterGroup
-      
+
       # @param [Hash] options
       #
       # @option options [String] :family
@@ -223,7 +223,7 @@ module Aws::RailsProvisioner
 
       def _default_family
         case @engine
-        when 'AURORA_POSTGRESQL' then 'aurora-postgresql9.6'
+        when 'AURORA_POSTGRESQL' then 'aurora-postgresql11'
         when 'AURORA_MYSQL' then 'aurora-mysql5.7'
         when 'AURORA' then 'aurora5.6'
         else
@@ -277,7 +277,7 @@ module Aws::RailsProvisioner
       when 'AURORA' then 'r5.large'
       else
         msg = 'Failed to locate a default instance type for :engine'\
-          ' provided, please provide :instance_type' 
+          ' provided, please provide :instance_type'
         raise Aws::RailsProvisioner::Errors::ValidationError, msg
       end
     end
@@ -289,7 +289,7 @@ module Aws::RailsProvisioner
       when 'AURORA' then 3306
       else
         msg = 'Failed to locate a default db port for :engine'\
-          ' provided, please provide :port' 
+          ' provided, please provide :port'
         raise Aws::RailsProvisioner::Errors::ValidationError, msg
       end
 

--- a/lib/aws-rails-provisioner/migration.rb
+++ b/lib/aws-rails-provisioner/migration.rb
@@ -20,7 +20,7 @@ module Aws::RailsProvisioner
     #   to `buildspec-db.yml` under root directory, using template under
     #   `buildspecs/`
     #
-    # @option options [String] :build_image default to codebuild `standard_1_0`
+    # @option options [String] :build_image default to codebuild `standard_4_0`
     #   full list of supported images see:
     #   https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-codebuild.LinuxBuildImage.html
     #
@@ -36,7 +36,7 @@ module Aws::RailsProvisioner
         options[:buildspec] = 'buildspec-db.yml'
       end
       unless options[:build_image]
-        options[:build_image] = 'standard_1_0'
+        options[:build_image] = 'standard_4_0'
       end
       # TODO envs support?
       super(options)

--- a/lib/aws-rails-provisioner/services.rb
+++ b/lib/aws-rails-provisioner/services.rb
@@ -1,20 +1,20 @@
 module Aws::RailsProvisioner
   # @api private
   class ServiceEnumerator
-    
+
     include Enumerable
 
     def initialize(options = {})
-      @configs = options || {}      
+      @configs = options || {}
     end
 
-    # @param [String] name 
+    # @param [String] name
     def [](name)
       if services.key?(name)
         services[name]
       else
-        msg = "unknown service #{identifier.inspect} under :services"
-        raise Aws::RailsProvisioner::ValidationError, msg
+        msg = "unknown service #{name} under :services"
+        raise Aws::RailsProvisioner::Errors::ValidationError, msg
       end
     end
     alias service []

--- a/lib/aws-rails-provisioner/views/fargate_stack.rb
+++ b/lib/aws-rails-provisioner/views/fargate_stack.rb
@@ -45,7 +45,6 @@ module Aws::RailsProvisioner
         base = [
           { abbr: 'ec2', value: 'ec2' },
           { abbr: 'ecs', value: 'ecs' },
-          { abbr: 'ecr', value: 'ecr' },
           { abbr: 'ecs_patterns', value: 'ecs-patterns' },
           { abbr: 'ecr_assets', value: 'ecr-assets' },
           { abbr: 'rds', value: 'rds' }

--- a/spec/fixtures/cdk/multi_service/rails-foo-fargate-stack.ts
+++ b/spec/fixtures/cdk/multi_service/rails-foo-fargate-stack.ts
@@ -1,7 +1,6 @@
 import cdk = require('@aws-cdk/core');
 import ec2 = require('@aws-cdk/aws-ec2');
 import ecs = require('@aws-cdk/aws-ecs');
-import ecr = require('@aws-cdk/aws-ecr');
 import ecs_patterns = require('@aws-cdk/aws-ecs-patterns');
 import ecr_assets = require('@aws-cdk/aws-ecr-assets');
 import rds = require('@aws-cdk/aws-rds');
@@ -56,7 +55,7 @@ export class RailsFooFargateStack extends cdk.Stack {
             instances: 2,
             parameterGroup: clusterParameterGroup
         });
-        const dbUrl = "mysql2://" + username + ":" + password + "@" + db.clusterEndpoint.socketAddress;
+        const dbUrl = "mysql2://" + username + ":" + password + "@" + db.clusterEndpoint.socketAddress + "/app_development";
         this.dbUrl = dbUrl;
 
         const asset = new ecr_assets.DockerImageAsset(this, 'ImageAssetBuild', {
@@ -68,8 +67,7 @@ export class RailsFooFargateStack extends cdk.Stack {
         const repoName = parts.slice(1, parts.length).join("/").split(":")[0];
         this.repoName = repoName;
 
-        const ecrRepo = ecr.Repository.fromRepositoryName(this, 'EcrRepo', repoName);
-        const image = ecs.ContainerImage.fromEcrRepository(ecrRepo);
+        const image = ecs.ContainerImage.fromDockerImageAsset(asset);
 
         // Fargate service
         const lbFargate = new ecs_patterns.ApplicationLoadBalancedFargateService(this, 'LBFargate', {
@@ -80,7 +78,7 @@ export class RailsFooFargateStack extends cdk.Stack {
               containerName: 'FargateTaskContainer',
               containerPort: 80,
               environment: {
-                  'DATABASE_URL': dbUrl, 
+                  'DATABASE_URL': dbUrl,
                   'PORT': '80',
               },
               enableLogging: true,

--- a/spec/fixtures/cdk/multi_service/rails-foo-pipeline-stack.ts
+++ b/spec/fixtures/cdk/multi_service/rails-foo-pipeline-stack.ts
@@ -106,7 +106,7 @@ export class RailsFooPipelineStack extends cdk.Stack {
                 }
             },
             environment:{
-                buildImage: codebuild.LinuxBuildImage.STANDARD_1_0
+                buildImage: codebuild.LinuxBuildImage.STANDARD_4_0
             },
             buildSpec: codebuild.BuildSpec.fromSourceFilename('buildspec-db.yml'),
             vpc: props.vpc,

--- a/spec/fixtures/cdk/multi_service/rails-no-db-fargate-stack.ts
+++ b/spec/fixtures/cdk/multi_service/rails-no-db-fargate-stack.ts
@@ -1,7 +1,6 @@
 import cdk = require('@aws-cdk/core');
 import ec2 = require('@aws-cdk/aws-ec2');
 import ecs = require('@aws-cdk/aws-ecs');
-import ecr = require('@aws-cdk/aws-ecr');
 import ecs_patterns = require('@aws-cdk/aws-ecs-patterns');
 import ecr_assets = require('@aws-cdk/aws-ecr-assets');
 import rds = require('@aws-cdk/aws-rds');
@@ -32,8 +31,7 @@ export class RailsNoDbFargateStack extends cdk.Stack {
         const repoName = parts.slice(1, parts.length).join("/").split(":")[0];
         this.repoName = repoName;
 
-        const ecrRepo = ecr.Repository.fromRepositoryName(this, 'EcrRepo', repoName);
-        const image = ecs.ContainerImage.fromEcrRepository(ecrRepo);
+        const image = ecs.ContainerImage.fromDockerImageAsset(asset);
 
         // Fargate service
         const lbFargate = new ecs_patterns.ApplicationLoadBalancedFargateService(this, 'LBFargate', {

--- a/spec/fixtures/cdk/no_db/rails-no-db-fargate-stack.ts
+++ b/spec/fixtures/cdk/no_db/rails-no-db-fargate-stack.ts
@@ -1,7 +1,6 @@
 import cdk = require('@aws-cdk/core');
 import ec2 = require('@aws-cdk/aws-ec2');
 import ecs = require('@aws-cdk/aws-ecs');
-import ecr = require('@aws-cdk/aws-ecr');
 import ecs_patterns = require('@aws-cdk/aws-ecs-patterns');
 import ecr_assets = require('@aws-cdk/aws-ecr-assets');
 import rds = require('@aws-cdk/aws-rds');
@@ -32,8 +31,7 @@ export class RailsNoDbFargateStack extends cdk.Stack {
         const repoName = parts.slice(1, parts.length).join("/").split(":")[0];
         this.repoName = repoName;
 
-        const ecrRepo = ecr.Repository.fromRepositoryName(this, 'EcrRepo', repoName);
-        const image = ecs.ContainerImage.fromEcrRepository(ecrRepo);
+        const image = ecs.ContainerImage.fromDockerImageAsset(asset);
 
         // Fargate service
         const lbFargate = new ecs_patterns.ApplicationLoadBalancedFargateService(this, 'LBFargate', {

--- a/spec/fixtures/cdk/single_service/rails-foo-fargate-stack.ts
+++ b/spec/fixtures/cdk/single_service/rails-foo-fargate-stack.ts
@@ -1,7 +1,6 @@
 import cdk = require('@aws-cdk/core');
 import ec2 = require('@aws-cdk/aws-ec2');
 import ecs = require('@aws-cdk/aws-ecs');
-import ecr = require('@aws-cdk/aws-ecr');
 import ecs_patterns = require('@aws-cdk/aws-ecs-patterns');
 import ecr_assets = require('@aws-cdk/aws-ecr-assets');
 import rds = require('@aws-cdk/aws-rds');
@@ -56,7 +55,7 @@ export class RailsFooFargateStack extends cdk.Stack {
             instances: 2,
             parameterGroup: clusterParameterGroup
         });
-        const dbUrl = "postgres://" + username + ":" + password + "@" + db.clusterEndpoint.socketAddress;
+        const dbUrl = "postgres://" + username + ":" + password + "@" + db.clusterEndpoint.socketAddress + "/app_development";
         this.dbUrl = dbUrl;
 
         const asset = new ecr_assets.DockerImageAsset(this, 'ImageAssetBuild', {
@@ -68,8 +67,7 @@ export class RailsFooFargateStack extends cdk.Stack {
         const repoName = parts.slice(1, parts.length).join("/").split(":")[0];
         this.repoName = repoName;
 
-        const ecrRepo = ecr.Repository.fromRepositoryName(this, 'EcrRepo', repoName);
-        const image = ecs.ContainerImage.fromEcrRepository(ecrRepo);
+        const image = ecs.ContainerImage.fromDockerImageAsset(asset);
 
         // Fargate service
         const lbFargate = new ecs_patterns.ApplicationLoadBalancedFargateService(this, 'LBFargate', {
@@ -80,7 +78,7 @@ export class RailsFooFargateStack extends cdk.Stack {
               containerName: 'FargateTaskContainer',
               containerPort: 80,
               environment: {
-                  'DATABASE_URL': dbUrl, 
+                  'DATABASE_URL': dbUrl,
               },
               enableLogging: true,
             },

--- a/spec/fixtures/cdk/single_service/rails-foo-pipeline-stack.ts
+++ b/spec/fixtures/cdk/single_service/rails-foo-pipeline-stack.ts
@@ -106,7 +106,7 @@ export class RailsFooPipelineStack extends cdk.Stack {
                 }
             },
             environment:{
-                buildImage: codebuild.LinuxBuildImage.STANDARD_1_0
+                buildImage: codebuild.LinuxBuildImage.STANDARD_4_0
             },
             buildSpec: codebuild.BuildSpec.fromSourceFilename('buildspec-db.yml'),
             vpc: props.vpc,

--- a/templates/fargate_stack.mustache
+++ b/templates/fargate_stack.mustache
@@ -93,7 +93,7 @@ export class {{stack_prefix}}FargateStack extends cdk.Stack {
             instances: {{instances}},
             parameterGroup: clusterParameterGroup
         });
-        const dbUrl = {{#postgres}}"postgres://"{{/postgres}}{{^postgres}}"mysql2://"{{/postgres}} + username + ":" + password + "@" + db.clusterEndpoint.socketAddress;
+        const dbUrl = {{#postgres}}"postgres://"{{/postgres}}{{^postgres}}"mysql2://"{{/postgres}} + username + ":" + password + "@" + db.clusterEndpoint.socketAddress + "/{{db_name}}";
         this.dbUrl = dbUrl;
         {{/db_cluster}}
 
@@ -106,8 +106,7 @@ export class {{stack_prefix}}FargateStack extends cdk.Stack {
         const repoName = parts.slice(1, parts.length).join("/").split(":")[0];
         this.repoName = repoName;
 
-        const ecrRepo = ecr.Repository.fromRepositoryName(this, 'EcrRepo', repoName);
-        const image = ecs.ContainerImage.fromEcrRepository(ecrRepo);
+        const image = ecs.ContainerImage.fromDockerImageAsset(asset);
 
         {{#fargate}}
         // Fargate service
@@ -122,7 +121,7 @@ export class {{stack_prefix}}FargateStack extends cdk.Stack {
               containerPort: {{container_port}},
               environment: {
                   {{#has_db}}
-                  'DATABASE_URL': dbUrl, 
+                  'DATABASE_URL': dbUrl,
                   {{/has_db}}
                   {{#envs}}
                   {{{.}}}
@@ -241,7 +240,7 @@ export class {{stack_prefix}}FargateStack extends cdk.Stack {
         {{/on_metric}}
         {{#on_request}}
         scaling.scaleOnRequestCount('FargateScalingOnRequest', {
-            requestsPerTarget: {{requests_per_target}},  
+            requestsPerTarget: {{requests_per_target}},
             targetGroup: target,
             {{#disable_scale_in}}
             disableScaleIn: true,

